shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/data/remote/apiclient/builder/HttpRequestBuilder.kt

```diff
@@ -2,7 +2,17 @@ package co.nimblehq.ic.kmm.suv.data.remote.apiclient.builder
 
 import co.nimblehq.ic.kmm.suv.BuildKonfig
 import io.ktor.client.request.*
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.jsonObject
 
 fun HttpRequestBuilder.path(path: String) {
     url(BuildKonfig.BASE_URL.plus(path))
 }
+
+inline fun <reified T> HttpRequestBuilder.setQueryParameters(parameters: T) {
+    val queryParameters = Json.encodeToJsonElement(parameters).jsonObject
+    for ((key, value) in queryParameters) {
+        parameter(key, value)
+    }
```

Comment:
I'd suggest adding `catch` in here to avoid encoding error

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/102/commits/62738814886749f933ffa56e0cd674c12a2b02ad